pulldown-cmark/src/firstpass.rs

```diff
@@ -724,8 +724,25 @@ impl<'a, 'b> FirstPass<'a, 'b> {
             let new_end = if has_trailing_content {
                 content_end
             } else {
+                let mut last_line_start = header_start;
+                loop {
+                    let next_line_start = last_line_start + scan_nextline(&bytes[last_line_start..content_end]);
+                    if next_line_start >= content_end {
+                        break;
+                    }
+                    let mut line_start = LineStart::new(&bytes[next_line_start..content_end]);
+                    if scan_containers(
+                        &self.tree,
+                        &mut line_start,
+                        self.options.has_gfm_footnotes(),
+                    ) != self.tree.spine_len()
+                    {
+                        break;
+                    }
+                    last_line_start = next_line_start + line_start.bytes_scanned();
+                }
```

Comment:
```suggestion
                if attrs.is_some() {
                    loop {
                        let next_line_start = last_line_start + scan_nextline(&bytes[last_line_start..content_end]);
                        if next_line_start >= content_end {
                            break;
                        }
                        let mut line_start = LineStart::new(&bytes[next_line_start..content_end]);
                        if scan_containers(
                            &self.tree,
                            &mut line_start,
                            self.options.has_gfm_footnotes(),
                        ) != self.tree.spine_len()
                        {
                            break;
                        }
                        last_line_start = next_line_start + line_start.bytes_scanned();
                    }
                }
```

Thanks for the better fix. If I'm understanding the issue correctly, this corner case only happens when heading attributes exist (because an empty line is never valid for setext header). The scanning part is only necessary when `attributes` is not empty.